main.js

```diff
@@ -0,0 +1,280 @@
+    }
+    dbgt(debugMessage.join(''))
+    originalErrFunction.apply(null, arguments)
+  }
+}())
+
+function Calculator() {
+  this.operands = []
+  this.tailIndex = 0
+  this.inputSize = 3.0
+  this.clearOrAllClear = 'all'
+  this.domElements = {
+    'Numbers': [
+      document.querySelector('.calculator-row:nth-child(5) > .calculator-field:nth-child(1) > button'),
+      document.querySelector('.calculator-row:nth-child(4) > .calculator-field:nth-child(1) > button'),
+      document.querySelector('.calculator-row:nth-child(4) > .calculator-field:nth-child(2) > button'),
+      document.querySelector('.calculator-row:nth-child(4) > .calculator-field:nth-child(3) > button'),
+      document.querySelector('.calculator-row:nth-child(3) > .calculator-field:nth-child(1) > button'),
+      document.querySelector('.calculator-row:nth-child(3) > .calculator-field:nth-child(2) > button'),
+      document.querySelector('.calculator-row:nth-child(3) > .calculator-field:nth-child(3) > button'),
+      document.querySelector('.calculator-row:nth-child(2) > .calculator-field:nth-child(1) > button'),
+      document.querySelector('.calculator-row:nth-child(2) > .calculator-field:nth-child(2) > button'),
+      document.querySelector('.calculator-row:nth-child(2) > .calculator-field:nth-child(3) > button'),
+    ],
+    'Operators': {
+      'Clear': document.querySelector('.calculator-row:nth-child(1) > .calculator-field:nth-child(1) > button'),
+      'Flip': document.querySelector('.calculator-row:nth-child(1) > .calculator-field:nth-child(2) > button'),
+      'Percentage': document.querySelector('.calculator-row:nth-child(1) > .calculator-field:nth-child(3) > button'),
+      'Divide': document.querySelector('.calculator-row:nth-child(1) > .calculator-field:nth-child(4) > button'),
+      'Multiply': document.querySelector('.calculator-row:nth-child(2) > .calculator-field:nth-child(4) > button'),
+      'Minus': document.querySelector('.calculator-row:nth-child(3) > .calculator-field:nth-child(4) > button'),
+      'Plus': document.querySelector('.calculator-row:nth-child(4) > .calculator-field:nth-child(4) > button'),
+      'Equals': document.querySelector('.calculator-row:nth-child(5) > .calculator-field:nth-child(3) > button'),
+      'Decimal': document.querySelector('.calculator-row:nth-child(5) > .calculator-field:nth-child(2) > button')
+    },
+    'CurrentInput': document.querySelector('.calculator-current-input'),
+    'History': document.querySelector('.calculator-history')
+  }
+
+  this.updateDisplayNode = function() {
+    var nodeToAdd = this.domElements.CurrentInput.firstChild
+
+    function lineHeightFromFontSize(fontSize, deltaSize) {
+      return ( 2.0 - (deltaSize*1.8) ).toString() // f(x) = 2 - 1.8x
+    }
+
+    if( !nodeToAdd ) {
+      nodeToAdd = document.createElement('p')
+      nodeToAdd.style.margin = 0
+      nodeToAdd.style.display = 'table-cell'
+      this.domElements.CurrentInput.appendChild(nodeToAdd)
+    }
```

Comment:
This is for generating a paragraph containing the numbers and operators that will be displayed in the calculator. I refactored, and was originally "deleting" the paragraphs but have since switched to just changing its textContent